cub/warp/warp_reduce.cuh

```diff
@@ -603,6 +603,74 @@ public:
     //@}  end member group
 };
 
+template <typename T, int LEGACY_PTX_ARCH>
+class WarpReduce<T, 1, LEGACY_PTX_ARCH>
+{
+private:
+  using _TempStorage = cub::NullType;
+
+  _TempStorage &temp_storage;
```

Comment:
Let's try to remove this, we shouldn't need it, and it would be nice for this to be an empty class.